core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1540,4 +1540,11 @@ private SqlLibraryOperators() {
           OperandTypes.DATETIME,
           SqlFunctionCategory.TIMEDATE);
 
+  @LibraryOperator(libraries = {SNOWFLAKE, ORACLE})
+  public static final SqlFunction BITNOT =
+          new SqlFunction("BITNOT",
+                  SqlKind.OTHER_FUNCTION,
+                  ReturnTypes.INTEGER, null,
+                  OperandTypes.family(SqlTypeFamily.NUMERIC),
```

Comment:
Kindly change SqlTypeFamily.NUMERIC to SqlTypeFamily.INTEGER, since BITNOT does not support the decimal argument.

Reply:
done.